eip/EIPS/eip-4337.md

```diff
@@ -624,6 +624,219 @@ Returns an array of the entryPoint addresses supported by the client. The first
 }
 ```
 
+#### * eth_chainId
+
+Returns EIP155 Chain ID.
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "eth_chainId",
+  "params": []
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "0x1"
+}
+```
+
+### debug Namespace
+
+#### * debug_bundler_clearState
+
+Clears the bundler mempool and reputation data of paymasters/accounts/factories/aggregators.
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_clearState",
+  "params": []
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "ok"
+}
+```
+
+#### * debug_bundler_dumpMempool
+
+Returns:
+
+`array` - Array of mempool entries objects, or an empty array.
+
+where each mempool entry has the following fields:
+
+`userOp` - UserOperation object. See `eth_sendUserOperation` for more details.
+`prefund` - The calculated prefund required for the user operation execution.
+`aggregator` - (optional) the aggregator used by the senderaccount.
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_dumpMempool",
+  "params": []
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": [
+    {
+      "userOp": {
+        sender, // address
+        nonce, // uint256
+        initCode, // bytes
+        callData, // bytes
+        callGasLimit, // uint256
+        verificationGasLimit, // uint256
+        preVerificationGas, // uint256
+        maxFeePerGas, // uint256
+        maxPriorityFeePerGas, // uint256
+        paymasterAndData, // bytes
+        signature // bytes
+      },
+      "prefund": "0x1234abcd",
```

Comment:
I think we're missing `entrypoint` here.  During periods of network upgrades, bundlers will participate in two mempools, one with the old entrypoint and one with the new one.  This is inevitable because the old one needs to be supported as long as some accounts haven't upgraded.  The `entrypoint` is also specified during eth_sendUserOperation, and is hashed into userOpHash.

Alternatively, add an optional `entrypoint` parameter to this call, since different entrypoints have different mempools.  This call should then only return the mempool of the requested entrypoint.  If not specified, use the primary one (the first one returned by `eth_supportedEntryPoints` as the EIP defines it to be the preferred one).

Reply:
yes, I agree an entryPoint is needed as parameter (like in sendUserOperation)